File: python/CommonPrimeDivisors.py
#https://codility.com/demo/results/trainingM3JK3F-58G/
def gcd(a, b):
    if b == 0:
        return a
    return gcd(b, a % b)
def solution(A, B):
    # write your code in Python 2.7
    cnt = 0
    for i in xrange(len(A)):
        a, b = A[i], B[i]
        g = gcd(a, b)
        while True:
            d = gcd(a, g)
            if 1 == d:
                break
            a /= d
        while True:
            d = gcd(b, g)
            if 1 == d:
                break
            b /= d
        cnt += 1 if a == 1 and b == 1 else 0
    return cnt

File: java/CommonPrimeDivisors.java
class SolutionCommonPrimeDivisors {
    private int gcd(int a, int b) {
        return 0 == b ? a : gcd(b, a % b);
    }
    public int solution(int[] A, int[] B) {
        int ans = 0, a = 0, b = 0, d = 0, dd = 0;
        for(int i = 0; i < A.length; ++i) {
            d = dd = gcd(A[i], B[i]);
            a = A[i];
            b = B[i];
            while(1 != d) {
                a /= d;
                d = gcd(a, d);
            }
            while(1 != dd) {
                b /= dd;
                dd = gcd(b, dd);
            }
            if(1 == a && 1 == b)++ans;
        }
        return ans;
    }
}
